lib/git-lfs-s3/application.rb

```diff
@@ -47,6 +49,151 @@ def protected!
       "Git LFS S3 is online."
     end
 
+    def valid_object?(object)
+      begin
+        valid = object[:size] >= 0
+      rescue
+        valid = false
+      end
+      begin        
+        if valid
+          oid = object[:oid].hex
+          valid = oid.size == 32 && object[:oid].size == 64
+        end
+      rescue
+        valid = false
+      end
+      valid
+    end
+
+    def object_download(authenticated, object, object_json)
+      # Format a single download object.
+      oid = object_json[:oid]
+      size = object_json[:size]
+      {
+        'oid' => oid,
+        'size' => size,
+        'authenticated' => authenticated,
+        'actions' => {
+          'download' => {
+            'href' => object.presigned_url(:get, :expires_in => 86400)
+          }
+        },
+        'expires_at' => DateTime.now.next_day.to_time.utc.iso8601
+      }
+    end
+
+    def object_upload(authenticated, object, object_json)
+      # Format a single upload object.
+      oid = object_json[:oid]
+      size = object_json[:size]
+      {
+        'oid' => oid,
+        'size' => size,
+        'authenticated' => authenticated,
+        'actions' => {
+          'upload' => {
+            'href' => object.presigned_url(:put, acl: 'public-read', :expires_in => 86400)
+          }
+        },
+        'expires_at' => DateTime.now.next_day.to_time.utc.iso8601
+      }
+    end
+
+    def object_error(error, message, object, object_json)
+      # Format a single error object.
+      {
+        'oid' => object_json[:oid],
+        'size' => object_json[:size],
+        'error' => {
+          'code' => error,
+          'message' => message
+        }
+      }
+    end
+
+    def download(authenticated, params)
+      # Handle git-lfs batch downloads.
+      objects = Array.new
+      params[:objects].each do |object_json|
+        object_json = indifferent_params object_json
```

Comment:
indifferent params is part of Sinatra. I used it so I could continue to use symbols instead of have a combination of strings and symbols. I imagine if I was more familiar with Sinatra and ruby there is a clever way to apply it everywhere simply. I'm open to changing it. https://github.com/sinatra/sinatra/blob/cab56ed7d64b9ffb6d4df48cb8d7d3153890f3ee/lib/sinatra/base.rb#L910